lib/shutdown.js

```diff
@@ -0,0 +1,62 @@
+var createTerminus = require('@godaddy/terminus').createTerminus;
+var _ = require('lodash');
+
+/**
+ * Clean up keystone server (e.g mongo connections) before shutdown
+ *
+ * @param {Object} keystone - keystone instance
+ * @returns {function} - which creates promise for the disconnection of mongoose
+ */
+function cleanupKeystone(keystone){
```

Comment:
recommend to use `async` for returning promise

Reply:
In this case, there's no obvious benefit from closure.
you can simply return promise by declare cleanupKeystone as a async function.
ex.
```
async function cleanupKeystone(ks) {
  // add log
  // mongoose.disconnect can return promise
  // note: express isn't fully support async/await, you may need to handle unexpected error / memory leak by try catch or safe await
  try {
    await ks.mongoose.disconnect()
    return
  } catch (err) {
    // error handle when disconnect fail
  }
}

Reply:
The reason why I used closure here was that `onSignal` accepted a function instead of a promise to do the cleanup.  See [here](https://github.com/godaddy/terminus/blame/main/README.md#L69). Do we have a better way to encapsulate the keystone without the closure?

Reply:
You're right, onSignal needs a function not a promise.
I'll resolve this command.